ospf6d/ospf6_top.c

```diff
@@ -652,21 +681,31 @@ DEFUN(no_ospf6_router_id,
 {
 	VTY_DECLVAR_CONTEXT(ospf6, o);
 	struct ospf6_area *oa;
-	struct listnode *node;
+	struct listnode *node, *nnode;
 
 	o->router_id_static = 0;
 
+	if (o->router_id == 0)
+		return CMD_SUCCESS;
+
 	for (ALL_LIST_ELEMENTS_RO(o->area_list, node, oa)) {
 		if (oa->full_nbrs) {
 			vty_out(vty,
 				"For this router-id change to take effect, save config and restart ospf6d\n");
 			return CMD_SUCCESS;
 		}
 	}
+
+	for (ALL_LIST_ELEMENTS(o->area_list, node, nnode, oa))
+		ospf6_area_disable(oa);
+
 	o->router_id = 0;
 	if (o->router_id_zebra.s_addr)
 		o->router_id = (uint32_t)o->router_id_zebra.s_addr;
 
+	for (ALL_LIST_ELEMENTS(o->area_list, node, nnode, oa))
+		ospf6_area_enable(oa);
```

Comment:
Same comment as above, we need to consider process level as well as interface well dynamic datas as well.